cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java

```diff
@@ -100,6 +103,19 @@ private void checkServiceResources(Set<String> problemKeys) {
     }
   }
 
+  private void checkBindAddress() {
+    // check if app.bind.address is loopback address
+    String appBindAddress = cConf.get(Constants.AppFabric.SERVER_ADDRESS);
+    try {
+      if (InetAddress.getByName(appBindAddress).isLoopbackAddress()) {
+        LOG.warn(Constants.AppFabric.SERVER_ADDRESS + " is set to " + appBindAddress + ". This is invalid in any " +
+                   "multinode Hadoop cluster.");
+      }
+    } catch (UnknownHostException e) {
+      LOG.warn("Unable to resolve " + Constants.AppFabric.SERVER_ADDRESS + ". " + e);
```

Comment:
check if the `warn` method allows specifying a Throwable as the last parameter.
If so, you could do:
`LOG.warn("...", e);`
That way, its not just the toString() of `e` that gets logged, but the entire stack trace also.